include/deal.II/particles/particle.h

```diff
@@ -0,0 +1,306 @@
+
+/**
+ * A namespace that contains all classes that are related to the particle
+ * implementation, in particular the fundamental <code>Particle</code> class.
+ */
+namespace Particles
+{
+  /**
+   * A namespace for all type definitions related to particles.
+   */
+  namespace types
+  {
+    /* Type definitions */
+
+#ifdef DEAL_II_WITH_64BIT_INDICES
+    /**
+     * The type used for indices of particles. While in
+     * sequential computations the 4 billion indices of 32-bit unsigned integers
+     * is plenty, parallel computations using hundreds of processes can overflow
+     * this number and we need a bigger index space. We here utilize the same
+     * build variable that controls the dof indices because the number
+     * of degrees of freedom and the number of particles are typically on the same
+     * order of magnitude.
+     *
+     * The data type always indicates an unsigned integer type.
+     */
+    typedef unsigned long long int particle_index;
+
+    /**
+     * An identifier that denotes the MPI type associated with
+     * types::global_dof_index.
+     */
+#  define PARTICLE_INDEX_MPI_TYPE MPI_UNSIGNED_LONG_LONG
+#else
+    /**
+     * The type used for indices of particles. While in
+     * sequential computations the 4 billion indices of 32-bit unsigned integers
+     * is plenty, parallel computations using hundreds of processes can overflow
+     * this number and we need a bigger index space. We here utilize the same
+     * build variable that controls the dof indices because the number
+     * of degrees of freedom and the number of particles are typically on the same
+     * order of magnitude.
+     *
+     * The data type always indicates an unsigned integer type.
+     */
+    typedef unsigned int particle_index;
+
+    /**
+     * An identifier that denotes the MPI type associated with
+     * types::global_dof_index.
+     */
+#  define PARTICLE_INDEX_MPI_TYPE MPI_UNSIGNED
+#endif
+  }
+
+  /**
+   * Base class of particles - represents a particle with position,
+   * an ID number and a variable number of properties. This class
+   * can be extended to include data related to a particle by the property
+   * manager.
+   *
+   *
+   * @ingroup Particle
+   *
+   */
+  template <int dim>
+  class Particle
+  {
+  public:
+    /**
+     * Empty constructor for Particle, creates a particle at the
+     * origin.
+     */
+    Particle ();
+
+    /**
+     * Constructor for Particle, creates a particle with the specified
+     * ID at the specified location. Note that deal.II
+     * does not check for duplicate particle IDs so the user must
+     * make sure the IDs are unique over all processes.
+     *
+     * @param new_location Initial location of particle.
+     * @param new_reference_location Initial location of the particle
+     * in the coordinate system of the reference cell.
+     * @param new_id Globally unique ID number of particle.
+     */
+    Particle (const Point<dim> &new_location,
```

Comment:
i think `new_` in variable names are a bit misleading, i would just call them `location` and `reference_location`.